crates/models/src/catalogs.rs

```diff
@@ -42,6 +42,37 @@ pub struct Catalog {
     pub tests: BTreeMap<Test, TestDef>,
 }
 
+#[derive(Serialize, Deserialize, Clone, Copy, Debug, JsonSchema, PartialEq, Eq)]
+#[serde(rename_all = "lowercase")]
+#[cfg_attr(
+    feature = "sqlx-support",
+    derive(sqlx::Type),
+    sqlx(type_name = "catalog_spec_type", rename_all = "lowercase")
+)]
+pub enum CatalogType {
+    Capture,
+    Collection,
+    Materialization,
+    Test,
+}
+
+/// Capability within the Estuary role-based access control (RBAC) authorization system.
+#[derive(
+    Serialize, Deserialize, Clone, Copy, Debug, JsonSchema, PartialEq, Eq, PartialOrd, Ord, Hash,
+)]
+#[serde(rename_all = "lowercase")]
+#[cfg_attr(
+    feature = "sqlx-support",
+    derive(sqlx::Type),
+    sqlx(type_name = "grant_capability", rename_all = "lowercase")
+)]
+pub enum Capability {
+    /// Note that the discriminants here align with those in the database type.
+    Read = 10,
```

Comment:
> It seems like sqlx is using the string representation, not the discriminant?

Yep, as far as I know

> The base 10 discriminants actuallly seem to align with the ones in the database, so the previous hex values were incorrect.

Yep, I change them because (for example) `read` is the 10th variant, not 0x10 = 16. AFAIK, nothing actually cares about these values however.